projects/meta_cl/experiments/dendrites.py

```diff
@@ -0,0 +1,309 @@
+            dendrite_sparsity,  # % of dendrites that are zero
+            dendrite_bias=None
+        )
+
+        self.prediction = nn.Sequential(
+            *self.conv_block(1, 256, 3, 2, 0),
+            *self.conv_block(256, 256, 3, 1, 0),
+            *self.conv_block(256, 256, 3, 2, 0),
+            nn.AdaptiveAvgPool2d(1),
+            nn.Flatten(),
+        )
+
+        self.modulation = nn.Sequential(
+            *self.conv_block(1, 256, 3, 2, 0),
+            *self.conv_block(256, 256, 3, 1, 0),
+            *self.conv_block(256, 256, 3, 2, 0),
+            nn.AdaptiveAvgPool2d(1),
+            nn.Flatten(),
+            nn.Linear(256, dim_context)
+        )
+
+        # apply Kaiming initialization
+        self.reset_params()
+
+    def reset_params(self):
+        # apply Kaiming initialization
+        for param in self.prediction.parameters():
+            if param.ndim > 1:
+                nn.init.kaiming_normal_(param)
+            else:
+                nn.init.zeros_(param)
+
+        for param in self.modulation.parameters():
+            if param.ndim > 1:
+                nn.init.kaiming_normal_(param)
+            else:
+                nn.init.zeros_(param)
+
+    @classmethod
+    def conv_block(cls, in_channels, out_channels, kernel_size, stride, padding):
+        return [
+            nn.Conv2d(
+                in_channels=in_channels, out_channels=out_channels,
+                kernel_size=kernel_size, stride=stride, padding=padding
+            ),
+            nn.ReLU(),
+        ]
+
+    @property
+    def named_fast_params(self):
+        """
+        Returns named params of adaption network, being sure to prepend
+        the names with "adaptation."
+        """
+        prepended = {}
+        for n, p in self.prediction.named_parameters():
+            n = "prediction." + n
+            prepended[n] = p
+        for n, p in self.gating_layer.named_parameters():
+            n = "gating_layer." + n
+            prepended[n] = p
+        return prepended
+
+    @property
+    def named_slow_params(self):
+        """
+        Returns named params of adaption network, being sure to prepend
+        the names with "representation."
+        """
+        prepended = {}
+        for n, p in self.modulation.named_parameters():
+            n = "modulation." + n
+            prepended[n] = p
+        return prepended
+
+    def forward(self, x):
+        mod = self.modulation(x)
+        pred = self.prediction(x)
+        out = self.gating_layer(pred, context=mod)
+        return out
+
+
+# Alternative to run on a single GPU
+def run_experiment(config):
+    exp = config.get("experiment_class")()
+    exp.setup_experiment(config)
+    print(f"Training started....")
+    while not exp.should_stop():
+        result = exp.run_epoch()
+        print(f"Accuracy: {result['mean_accuracy']:.4f}")
+    print(f"....Training finished")
+
+
+# Simple Omniglot Experiment
+metacl_base = dict(
```

Comment:
Can you remove the unused configs?

Reply:
There are no unused configs in this file. Each one is copying from the previous one. Notice the only one exposed though is the metacl_dendrites (ps: I didn't want it to have dependencies on other files)

Reply:
Oh I see